src/build-images-handler.ts

```diff
@@ -0,0 +1,170 @@
+    console.error('Error fetching previous target_oid:', error);
+    return null;
+  }
+}
+
+export async function updateFilesWithNewOid(
+  octokit: Octokit,
+  filePaths: string[],
+  previousTargetOid: string,
+  targetOid: string,
+  branchName: string,
+) {
+  const d = debug(`roller/github:updateFilesWithNewOid`);
+  let updatedAny = false;
+
+  for (const filePath of filePaths) {
+    try {
+      const { data: file } = await octokit.rest.repos.getContent({
+        ...REPOS.electron,
+        path: filePath,
+      });
+
+      if (!('content' in file)) {
+        throw new Error(`Incorrectly received array when fetching content for ${filePath}`);
+      }
+
+      const fileContent = Buffer.from(file.content, 'base64').toString('utf-8');
+      const match = fileContent.match(previousTargetOid);
+      if (!match) {
+        d(`No match found for ${filePath}`);
+        continue;
+      }
+
+      const currentSha = match[0];
+      if (currentSha === targetOid) {
+        d(`No update needed for ${filePath}`);
+        continue;
+      }
+
+      d(`Updating ${filePath} from ${currentSha} to ${targetOid}`);
+      const newContent = fileContent.replace(currentSha, targetOid);
+      await octokit.rest.repos.createOrUpdateFileContents({
+        ...REPOS.electron,
+        path: filePath,
+        content: Buffer.from(newContent).toString('base64'),
+        message: `chore: bump build image tag in ${filePath} to ${targetOid.slice(0, 7)}`,
+        sha: file.sha,
+        branch: branchName,
+      });
+      updatedAny = true;
+    } catch (error) {
+      d(`Failed to update ${filePath}: ${error.message}`);
+    }
+  }
+
+  return updatedAny;
+}
+
+export async function prepareGitBranch(octokit: Octokit, branchName: string, mainBranch: string) {
+  const d = debug(`roller/github:prepareGitBranch`);
+
+  const { data: branch } = await octokit.rest.repos.getBranch({
+    ...REPOS.electron,
+    branch: mainBranch,
+  });
+
+  const sha = branch.commit.sha;
+  const shortRef = `heads/${branchName}`;
+  const ref = `refs/${shortRef}`;
+
+  // Clean up any orphan refs
+  d(`Checking that no orphan ref exists from a previous roll`);
+  try {
+    const maybeOldRef = await octokit.rest.git.getRef({ ...REPOS.electron, ref: shortRef });
+    if (maybeOldRef.status === 200) {
+      d(`Found orphan ref ${ref} with no open PR - deleting`);
+      await octokit.rest.git.deleteRef({ ...REPOS.electron, ref: shortRef });
+      await new Promise<void>((r) => setTimeout(r, 2000));
+    }
+  } catch (error) {
+    d(`No orphan ref exists at ${ref} - proceeding, `, error);
+  }
+
+  return { ref, shortRef, branchName, sha };
+}
+
+export async function handleBuildImagesCheck(payload: RegistryPackagePublishedEvent) {
+  const d = debug(`roller/github:handleBuildImagesCheck`);
+  const octokit = await getOctokit();
+
+  const { target_oid } = payload.registry_package.package_version;
+  const previousTargetOid = await getPreviousTargetOid(payload);
```

Comment:
Is it always gonna be the exact previous oid, maybe we should get a list of like the last 30 or something and look to replace any of the options